test/codegen/delete_translator_test.cpp

```diff
@@ -0,0 +1,222 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+//
+// delete_translator_test.cpp
+//
+// Identification: test/codegen/delete_translator_test.cpp
+//
+// Copyright (c) 2015-17, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "catalog/catalog.h"
+#include "codegen/codegen_test_util.h"
+#include "common/harness.h"
+#include "expression/conjunction_expression.h"
+#include "expression/operator_expression.h"
+
+namespace peloton {
+namespace test {
+
+//===----------------------------------------------------------------------===//
+// This class contains code to test code generation and compilation of insert
+// plans. All tests use a test table with the following schema:
+//
+// +---------+---------+---------+-------------+
+// | A (int) | B (int) | C (int) | D (varchar) |
+// +---------+---------+---------+-------------+
+//
+//===----------------------------------------------------------------------===//
+
+class DeleteTranslatorTest : public PelotonCodeGenTest {
+ public:
+  DeleteTranslatorTest() : PelotonCodeGenTest(), num_rows_to_insert(64) {
+    LoadTestTable(TestTableId(), num_rows_to_insert);
+  }
+
+  size_t getCurrentTableSize() {
+    planner::SeqScanPlan scan{&GetTestTable(TestTableId()), nullptr, {0, 1}};
+    planner::BindingContext context;
+    scan.PerformBinding(context);
+
+    codegen::BufferingConsumer buffer{{0, 1}, context};
+    CompileAndExecute(scan, buffer, reinterpret_cast<char*>(buffer.GetState()));
+    return buffer.GetOutputTuples().size();
+  }
+
+  // delete all entries in table and then repopulate it.
+  void reloadTable() {
+    std::unique_ptr<planner::DeletePlan> delete_plan{
```

Comment:
You can remove this method and just load the test table at the start of each test, because this contains functionality that's under test.